### ISSUE TITLE
Randomness-based obfuscation

### ISSUE THREAD

**Issue by nacho00112** [billythegoat356/Hyperion#35]
Due to indeterminated results, this may stop deobfuscators

**Comment by billythegoat356** [comment]
hey, what do you mean?

**Comment by nacho00112** [comment]
Going into more detail, at this point you may want to use ast obfuscation instead of token obfuscation:

1. Randomized false control flow
    
    You could generate false control flow and mix the real code logic with it, everything in a random and confusing way,
    A database (json?) May be handy here, like, with a list with variable names to use in the generation
    And other stuff you could need, this will probably require a more sophisticated analysis of
    The python code, to create false control flow that looks real and that provides great camouflage
    To the real code logic, that's why I mentioned ast

2. File size control

    That is, that you can control how much false control flow you want in the code,
     The more, the bigger the file, that might increase obfuscation level, or you could determine this automatically based on real code's size

3. Source code directories to one single file converter

    There are already libraries that do this, just imagine a large codebase of hundred of files 
     Compressed in one simple python file, that would be a plus in the obfuscation,
    This may also need some randomization to avoid deobfuscators reversing this

**Comment by billythegoat356** [comment]
I see what you mean
This is a nice idea, but sadly i made this obfuscator before learning about AST
It's not in my projects anymore, but it's definitely a smart way of stopping deobfuscators!

**Comment by nacho00112** [comment]
Yeah, because once the obfuscation finish, the values of the random variables are lost,
So, you prefer considering this in a new project instead of adding to hyperion
Or you do no longer want to create obfuscators?

**Comment by billythegoat356** [comment]
I am no longer active on github and public projects like that. I am mostly doing personal projects
So this idea will most likely not happen  :(

**Comment by nacho00112** [comment]
That's sad, it was a quite famous project I liked it, if you ever have some code in your projects to publish you might want to obfuscate it, if that happens I hope you come back

**Comment by billythegoat356** [comment]
Maybe in the future 🙏
